src/Machete.HL7.Tests/AdvancedQueryTests.cs

```diff
@@ -0,0 +1,325 @@
+﻿namespace Machete.HL7.Tests
+{
+    using System;
+    using HL7Schema.V26;
+    using NUnit.Framework;
+    using Testing;
+
+
+    [TestFixture]
+    public class AdvancedQueryTests :
+        MacheteHL7TestContext<Version26>
+    {
+        [Test]
+        public void Should_be_able_to_query_blocks()
+        {
+            const string message = @"MSH|^~\&|NextGen|^DOSC|RH|18779|20130405125146269||ORM^O01|1999077678|P|2.3|||AL|AL
+NTE|1||KOPASD
```

Comment:
i can't let this in as long as this test data is here. It needs to be hand crafted.